StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Link/PayWithNativeLinkController.swift

```diff
@@ -16,23 +16,54 @@ import UIKit
 @available(macCatalystApplicationExtension, unavailable)
 final class PayWithNativeLinkController {
 
+    enum Mode {
+        case full
+        case paymentMethodSelection
+    }
+
+    enum CompletionResult {
```

Comment:
Introduced this as a way to allow two different types of result, since the full flow will return a `PaymentSheetResult` and the payment method selection flow will return a `LinkConfirmOption`.